src/NuGet.Clients/NuGet.VisualStudio.Common/ServiceLocator.cs

```diff
@@ -96,23 +92,33 @@ public static void InitializePackageServiceProvider(IServiceProvider provider)
             // and so this method can RPC into main thread. Switch to main thread explictly, since method has STA requirement
             await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            return await GetGlobalServiceFreeThreadedAsync<TService, TInterface>();
+            if (PackageServiceProvider != null)
+            {
+                var result = await PackageServiceProvider.GetServiceAsync(typeof(TService));
+                var service = result as TInterface;
+                if (service != null)
+                {
+                    return service;
+                }
+            }
+
+            return Package.GetGlobalService(typeof(TService)) as TInterface;
```

Comment:
Should we add telemetry for when this or [this](https://github.com/NuGet/NuGet.Client/pull/2898/files#diff-6b14c4c96925a5094908e8cb833d59b4R121) is called?

Reply:
What would be trying to learn? 
It would just tell us that the service in question is not async. 

Maybe something else?

Reply:
@dtivel 

I'll merge this now, so it can runt through the nightly validations, but i'll be hpapy to discuss further improvements to this.